Tangem/Modules/Main/MainViewModel.swift

```diff
@@ -304,13 +304,17 @@ final class MainViewModel: ObservableObject {
     private func bind() {
         $selectedCardIndex
             .dropFirst()
-            .sink { [weak self] newIndex in
-                guard let userWalletId = self?.pages[newIndex].id else {
+            .withWeakCaptureOf(self)
+            .sink { viewModel, newIndex in
+                let pagesIndices = 0 ..< viewModel.pages.count
+                guard pagesIndices.contains(newIndex) else {
                     return
                 }
 
+                let userWalletId = viewModel.pages[newIndex].id
+
```

Comment:
```suggestion
                guard let userWalletId = viewModel.pages[safe: newIndex].id else {
                    return
                }
```
а там есть уже extension под это